src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -273,7 +273,7 @@ export class WorkbenchEditorService implements IWorkbenchEditorService {
 		}
 
 		let input: ICachedEditorInput;
-		if (resource.scheme === network.Schemas.file) {
+		if (resource.scheme === network.Schemas.file || resource.scheme === 'ftp') {
 			input = this.fileInputFactory.createFileInput(resource, encoding, instantiationService);
```

Comment:
@bpasero what is the different between a `FileInput` and `ResourceInput`? Is this just about support save/dirty? Should this become a (editor) document property? E.g. when reading a file we mark it as readonly or not?

Reply:
@jrieken yeah the file input was first and then resource input was added later, mainly for readonly documents. I think ideally untitled and file inputs would merge into resource input and we have a way to set a scheme as writeable. 

I think the real work is not about merging the inputs but taking the code we have in the file editor model over to the "editable" resource model. Because that code contains the logic for handling hot exit and saving. That work is captured in https://github.com/Microsoft/vscode/issues/29025